src/App.tsx

```diff
@@ -1,14 +1,75 @@
-/* eslint-disable max-len */
-import React from 'react';
+/* eslint-disable no-nested-ternary */
+import React, { useState, useEffect, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilteredBy } from './types/Filter';
+import { getTodos } from './api';
+
+const filteredTodos = (todos: Todo[], query: string, filterBy: FilteredBy) => {
+  const readyQuery = query ? query.toLowerCase().trim() : '';
+
+  const readyTodos = todos.filter(todo => {
+    const todoTitle = todo.title.toLowerCase().trim();
+    const isTitleMatched = todoTitle.includes(readyQuery);
+    const isActiveFilter = filterBy === FilteredBy.ACTIVE && !todo.completed;
+    const isCompletedFilter = filterBy === FilteredBy.COMPLETED
+        && todo.completed;
+    const isAllFilter = filterBy === FilteredBy.ALL;
+
+    return (
+      isTitleMatched
+      && (isActiveFilter
+        || isCompletedFilter
+        || isAllFilter
+      )
+    );
+  });
+
+  return readyTodos;
+};
```

Comment:
You solution is unredable. Look at this
```suggestion
const getVisibleTodos = (todos: Todo[], query: string, filterBy: FilteredBy) => {
 
  const visibleTodos = filterBy !== FilteredBy.ALL 
    ? todos.filter(todo => {
      if( FilteredBy.ACTIVE) {
        return !todo.completed;
      }
      
      return todo.completed;
    });
    : todos;
    
  if (query) {
    const readyQuery = query.toLowerCase().trim();
    
    return visibleTodos.filter(todo => todo.title.toLowerCase().includes(readyQuery));
  }
  
  return visibleTodos;
}
```